src/index.ts

```diff
@@ -92,3 +92,56 @@ export function render(
     })
     .catch(error => showError(el, error));
 }
+
+// optionally define a Jupyter Widget if the plugin is available available
+export const VegaWidget = import('@jupyter-widgets/base')
+.then(widgets => {
+  return class VegaWidget extends widgets.DOMWidgetView {
+    view: any
+
+    render() {
+          const reembed = () => {
+              this.view = null;
+              const spec = JSON.parse(this.model.get('_spec_source'));
+
+              if(spec == null) {
+                  return
+              }
+
+              vegaEmbed(this.el, spec)
+              .then(({view}) => {
+                  this.view = view;
+              })
+              .catch(err => console.error(err));
+          };
+
+          this.model.on('change:spec_source', reembed);
+          this.model.on('msg:custom', ev => {
+              if(ev.type != 'update') {
+                  return;
+              }
+              if(this.view == null) {
+                  console.error('no view attached to widget');
+                  return
+              }
+
+              const filter = new Function('datum', 'return (' + (ev.remove || 'false') + ')');
+              const newValues = ev.insert || [];
+
+              const changeSet = (
```

Comment:
why not just 

```ts
const changeSet = this.view.changeset().insert(newValues).remove(filter)
```

You can access `vegaEmbed.vega.changeset()` as well but this is fine.